pkg/k8s/node.go

```diff
@@ -52,8 +52,8 @@ func ParseNode(k8sNode *v1.Node, source node.Source) *node.Node {
 			continue
 		}
 		na := node.Address{
-			AddressType: addr.Type,
-			IP:          ip,
+			Type: node.AddressType(addr.Type),
```

Comment:
[1] And here, instead of doing a direct cast we would create a function to parse from a k8s address type to a `pkg/addressing` type and log an error if we don't recognize the type. BTW, are we using the Type for anything internally?

Reply:
Good idea re: the parsing function. Coming in next upload :) 

In response to your last question, yes, we do have some logic depending on whether the type is `NodeInternalIP`, `NodeExternalIP`, or something else within `pkg/node/node.go:getNodeIP`.